aiida/cmdline/commands/cmd_import.py

```diff
@@ -157,14 +215,64 @@ def cmd_import(archives, webpages, group, extras_mode_existing, extras_mode_new,
                     extras_mode_existing=ExtrasImportCode[extras_mode_existing].value,
                     extras_mode_new=extras_mode_new,
                     comment_mode=comment_mode)
-            except exceptions.IncompatibleArchiveVersionError as exception:
-                crit_message = '{} cannot be imported.\n' \
-                    'Download the archive file and run `verdi export migrate` to update it.\n{}'.format(
-                    archive, exception)
-                echo.echo_critical(crit_message)
+            except exceptions.IncompatibleArchiveVersionError:
+                echo.echo_warning('{} cannot be imported.'.format(archive))
+                if non_interactive:
+                    migrate_archive = manual_migration
+                else:
+                    migrate_archive = click.confirm(
+                        "Do you want to try and migrate {} to the newest export file version?\n"
+                        "Note: A temporary local file will be generated.".format(archive),
+                        default=manual_migration,
+                        abort=True)
             except Exception:
                 echo.echo_error('an exception occurred while importing the archive {}'.format(archive))
                 echo.echo(traceback.format_exc())
-                click.confirm('do you want to continue?', abort=True)
+                if not non_interactive:
+                    click.confirm('do you want to continue?', abort=True)
             else:
                 echo.echo_success('imported archive {}'.format(archive))
+
+            # Migrate archive if needed and desired
+            if migrate_archive:
+                migrate_archive = False
```

Comment:
if beer then: no beer ;-)

I guess this line is anyhow not needed...

Reply:
Yeah, it looks weird. But if my analysis of the for-loop and everything is correct, I believe it is needed.

We can have multiple archives, where one may have to be migrated (setting `migrate_archive = True`). Then a later archive, in the same for-loop, will not have to be migrated, `migrate_archive` is still `True`, so it will try to migrate it and fail.

Reply:
Ok, then rather put `migrate_archive=False` at the beginning of the loop

Reply:
True! Thanks.

Reply:
Done.